src/cortxfs/cortxfs_ops.c

```diff
@@ -637,67 +649,68 @@ int cfs_rename(struct cfs_fs *cfs_fs, cfs_cred_t *cred,
 	dassert((*sino_dir != *dino_dir || strcmp(sname, dname) != 0));
 	dassert(cfs_fs);
 
-	str256_from_cstr(k_sname, sname, strlen(sname));
-	str256_from_cstr(k_dname, dname, strlen(dname));
+	/* TODO:Temp_FH_op - to be removed
+	 * Should get rid of creating and destroying FH operation in this
+	 * API when caller pass the valid FH instead of inode number
+	 */
+	RC_WRAP_LABEL(rc, out, cfs_fh_from_ino, cfs_fs, sino_dir, &sdir_fh);
+	RC_WRAP_LABEL(rc, out, cfs_fh_from_ino, cfs_fs, dino_dir, &ddir_fh);
 
-	rename_inplace = (*sino_dir == *dino_dir);
+	sdir_ino = cfs_fh_ino(sdir_fh);
+	sdir_stat = cfs_fh_stat(sdir_fh);
+	sdir_id = cfs_node_id_from_fh(sdir_fh);
+	ddir_ino = cfs_fh_ino(ddir_fh);
+	ddir_stat = cfs_fh_stat(ddir_fh);
+	ddir_id = cfs_node_id_from_fh(ddir_fh);
 
-	RC_WRAP_LABEL(rc, out, cfs_access, cfs_fs, cred, sino_dir,
+	rename_inplace = (*sdir_ino == *ddir_ino);
+
+	RC_WRAP_LABEL(rc, out, cfs_access_check, cred, sdir_stat,
 		      CFS_ACCESS_DELETE_ENTITY);
 
 	if (!rename_inplace) {
-		RC_WRAP_LABEL(rc, out, cfs_access, cfs_fs, cred, dino_dir,
+		RC_WRAP_LABEL(rc, out, cfs_access_check, cred, ddir_stat,
 			      CFS_ACCESS_CREATE_ENTITY);
 	}
 
-	if (psrc) {
-		sino = *psrc;
-	} else {
-		RC_WRAP_LABEL(rc, out, cfs_lookup, cfs_fs, cred, sino_dir, sname,
-			      &sino);
-	}
+	RC_WRAP_LABEL(rc, out, cfs_fh_lookup, cred, sdir_fh, sname, &src_fh);
```

Comment:
Suggestion: I guess it would be good idea to check fh invariant after every lookup. This way, a corrupted fh would be caught before causing sigsegv or memory corruptions later.

Reply:
`cfs_fh_lookup`  will do invariant check before returning a validi `fh` so not needed to do it again here.